sdk/personalizer/Azure.AI.Personalizer/src/Models/DecisionContext.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Azure.AI.Personalizer
+{
+    /// <summary> The Decision Context. </summary>
+    public class DecisionContext
+    {
+        /// <summary> The Decision Context. </summary>
+        public DecisionContext()
+        {
+        }
+
+        /// <summary> Initializes a new instance of DecisionContext. </summary>
+        /// <param name="rankRequest"> Personalizer Rank Options </param>
+        public DecisionContext(PersonalizerRankOptions rankRequest)
+        {
+            List<string> jsonFeatures = rankRequest.ContextFeatures.Select(f => JsonConvert.SerializeObject(f)).ToList();
+            this.SharedFromUrl = jsonFeatures;
+
+            this.Documents = rankRequest.Actions
+                .Select(action =>
+                {
+                    List<string> jsonFeatures = action.Features.Select(f => JsonConvert.SerializeObject(f)).ToList();
+
+                    var doc = new DecisionContextDocument
+                    {
+                        ID = action.Id,
+                        JSON = jsonFeatures,
+                    };
+
+                    return doc;
+                }).ToArray();
+        }
+
+        /// <summary> Properties from url </summary>
+        [JsonProperty("FromUrl", NullValueHandling = NullValueHandling.Ignore)]
+        [JsonConverter(typeof(JsonRawStringListConverter))]
+#pragma warning disable CA2227 // Collection properties should be read only
```

Comment:
[nit] this should be moved to the project file. For my knowledge, why do we need SharedFromUrl to not be read only?